dev/Repeater/SelectionModelChildrenRequestedEventArgs.cpp

```diff
@@ -33,7 +33,7 @@ winrt::IndexPath SelectionModelChildrenRequestedEventArgs::SourceIndex()
         throw winrt::hresult_error(E_FAIL, L"SourceIndex can only be accesed in the ChildrenRequested event handler.");
     }
```

Comment:
This isn't used anymore right?

Reply:
The check is to disallow caching the args object and calling these properties on them later. We decided to not allow that because  allocating a new args object for every event which can be a lot of allocations making it a performance issue. Also, not throwing here can cause for some confusing behavior and an error right at the point of usage will help someone from going down that rabbit hole. 

We do not need to use m_sourceNode to do this check though, we can remove it and use a bool or one of the other properties to gate the api call. A comment here describing the reason is warranted i think.

---
In reply to: [377802966](https://github.com/microsoft/microsoft-ui-xaml/pull/1963#discussion_r377802966) [](ancestors = 377802966)

Reply:
Replaced m_sourceNode with a bool in order to act as a sentinel